pandas/io/common.py

```diff
@@ -419,13 +463,19 @@ class BytesZipFile(zipfile.ZipFile, BytesIO):  # type: ignore
     bytes strings into a member of the archive.
     """
     # GH 17778
-    def __init__(self, file, mode, compression=zipfile.ZIP_DEFLATED, **kwargs):
+    def __init__(self, file, mode, compression=zipfile.ZIP_DEFLATED,
+                 archive_name: Union[str, zipfile.ZipInfo, None] = None,
```

Comment:
Where is zipfile.ZipInfo coming from here?

Reply:
@WillAyd A `ZipFile.writestr` can take either a `ZipInfo` or a string as the archived file name `zinfo_or_arcname`,  which `archive_name` is passed to, so I figured that type may as well be allowed here.

Reply:
Right but internally we are only calling with a str right?

Reply:
`archive_name` is whatever is passed at `compression['archive_name']`, so it _could_ be a `ZipInfo`

Reply:
Can you provide a use case for when this would happen? If allowable then should be reflected in the docstring and with a test

Reply:
I can tell that file comments and the file date can be set by passing a `ZipInfo`. Not sure if this is an important enough functionality to include - what do you think?

Reply:
I'm not sure I follow what you are saying but unless this is a feature you think should be documented and supported it should be removed from the annotation

Reply:
Got it - I'll remove that type from the annotation.